domain-eval/evaluation_task.py

```diff
@@ -0,0 +1,95 @@
+import argparse
+import numpy as np
+
+from lighteval.tasks.task import LightevalTaskConfig
+from lighteval.tasks.task_utils import Doc
+from lighteval.metrics.metrics import SampleLevelMetric, MetricCategory, MetricUseCase
+
+################################################################################
+# Script Parameters
+################################################################################
+
+parser = argparse.ArgumentParser(
+    description="Define a LightEval task for exam questions."
+)
+parser.add_argument(
+    "--hf_repo",
+    type=str,
+    default="burtenshaw/exam_questions",
+    help="Hugging Face dataset repository ID",
+)
+parser.add_argument(
+    "--hf_subset",
+    type=str,
+    default="default",
+    help="Subset of the dataset to use",
+)
+parser.add_argument(
+    "--evaluation_split",
+    type=str,
+    default="train",
+    help="Split of the dataset to use for evaluation",
+)
+
+args = parser.parse_args()
+
+################################################################################
+# Define the prompt function
+################################################################################
+
+
+def prompt_fn(line, task_name: str = None):
+    """Converts a dataset line to a Doc object for evaluation."""
+    return Doc(
+        task_name=task_name,
+        query=line["question"],
+        choices=[
+            f" {line['answer_a']}",
+            f" {line['answer_b']}",
+            f" {line['answer_c']}",
+            f" {line['answer_d']}",
+        ],
+        gold_index=["answer_a", "answer_b", "answer_c", "answer_d"].index(
+            line["correct_answer"]
+        ),
+        instruction="Choose the correct answer for the following exam question:",
+    )
+
+
+################################################################################
+# Define the custom metric
+################################################################################
+
+custom_metric = SampleLevelMetric(
```

Comment:
Are you sure you need a custom metric for this? Looks like a perfect exact match

Reply:
You're right it doesn't. I included it for the completeness of the tutorial, in case the reader sets up a task that does require one. Emulating the lighteval's custom task guide [here](https://github.com/huggingface/lighteval/wiki/Adding-a-Custom-Task). Do you think it's over-complexity?

Reply:
A bit yes

Reply:
But at the same time it's interesting, maybe leave it as such and provide the link to the other metrics if people need?

Reply:
Will do